### PR TITLE
feat: update Terser to version 5

### PR DESCRIPTION
**Pull request by chicoxyzzy**
This PR contains a:

- [ ] **bugfix**
- [x] new **feature**
- [ ] **code refactor**
- [x] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

Terser 5 release =)

### Breaking Changes

`minify` and `transform` are async now

### CODE REVIEW

File: test/__snapshots__/terserOptions-option.test.js.snap.webpack4 [hunk 1]
```diff
@@ -662,9 +662,4 @@ Object {
 
 exports[`terserOptions option should match snapshot for the "warnings" option with the "true" value: errors 1`] = `Array []`;
 
-exports[`terserOptions option should match snapshot for the "warnings" option with the "true" value: warnings 1`] = `
-Array [
-  "Terser Plugin: Dropping unreachable code [main.js:91,41]",
-  "Terser Plugin: Dropping unused function foo [main.js:91,9]",
-]
-`;
+exports[`terserOptions option should match snapshot for the "warnings" option with the "true" value: warnings 1`] = `Array []`;
```
Comment (chicoxyzzy) [inline test/__snapshots__/terserOptions-option.test.js.snap.webpack4:665]: I'm not sure why these arrays are empty now 🤔

Comment (chicoxyzzy) [reply test/__snapshots__/terserOptions-option.test.js.snap.webpack4:665]: warnings are deprecated https://github.com/terser/terser/commit/71dc3456be968ddce2281b93beb4079e609e28ef

Comment (evilebottnawi) [reply test/__snapshots__/terserOptions-option.test.js.snap.webpack4:665]: It is very bad, very

**Comment by evilebottnawi** [comment]
I will do the next major release tomorrow with `terser@5`, if you do rebase, it helps me, thanks!

**Comment by chicoxyzzy** [comment]
tests are flaky

**Comment by evilebottnawi** [comment]
github actions bugs on mac os, reported many time, but no luck

**Comment by chicoxyzzy** [comment]
oh, I didn't notice lint error. will fix it now

**Comment by evilebottnawi** [comment]
I already do new PR, no need removing `warnings`, it is `warnings` from webpack, anyway big thanks for the PR

**Comment by chicoxyzzy** [comment]
that's why I created the second PR which could be simply dropped

**Comment by evilebottnawi** [comment]
If you remove it then I can merge your work and make a little more changes

**Comment by evilebottnawi** [comment]
Or we can merge it https://github.com/webpack-contrib/terser-webpack-plugin/pull/291 :smile:

**Comment by TrySound** [comment]
Better merge the one which is first.

**Comment by chicoxyzzy** [comment]
it's dropped

**Comment by evilebottnawi** [comment]
Thanks

**Pull request merged**